tests/fsharp/Compiler/Conformance/BasicGrammarElements/BasicConstants.fs

```diff
@@ -172,3 +174,18 @@ let x14 = 0o5_2
 if x14 <> 0o52 then failwith "Wrong parsing"
 printfn "%A" x14
             """
+    [<Test>]
+    let ``float without dot``() = 
+        CompilerAssert.CompileExeAndRun
```

Comment:
In this case since we're not verifying the output in any way, `Compile` is probably sufficient. We'll want to also test that things like `1.0f = `1f`.

Reply:
Added missing test and upgraded existing tests.